index.bs

```diff
@@ -274,6 +275,12 @@ for plans to make {{console/timeEnd()}} and {{console/timeLog()}} formally repor
 console when a given |label| does not exist in the associated <a>timer table</a>.
 </p>
 
+<h4 id="timestamp" oldids="timestamp-label,dom-console-timestamp" method for="console">timeStamp(|label|)</h4>
+
+1. If the developer is recording a performance trace, add a single marker to the performance trace with the |label|.
```

Comment:
im slightly concerned that this is the first time we've mentioned anything like "performance trace" in this spec.  this feels a bit limiting to me (e.g. what if instead of showing a marker we'd like to put it in a table?  how does this work with non-GUI environments?  etc.).  is there a way we can make this a bit more generic about just "capture a timestamp (relative to XYZ) for use in an implementation-defined way"?

Reply:
Hmm, at first I strongly agreed, but when I looked into what [Node](https://nodejs.org/api/console.html#consoletimestamplabel) and [Deno](https://docs.deno.com/api/web/~/Console.timeStamp) do, they specifically mention the method is a noop unless hooked up to an inspector environment with a timeline UI. Maybe this isn't too bad to codify directly in the spec, since it matches reality with every single implementation, including non-browser ones.

But if people feel strongly, we can avoid that, making the normative text more vague, and adding some non-normative text mirroring what common developer documentation mentions about this API — namely that in browser/UI environments, it generally targets a performance-ey panel of sorts.

Reply:
We can also make this more vague, but I thought it'd be useful to capture reasonable expectations for the API in the spec, so developers know that for conforming implementations

1. this is a no-op when no trace is being recorded, and
2. when a trace is being recorded, calls to `console.timeStamp()` will leave a mark in the trace in some form.

Reply:
Yeah, I think I'm OK with what's currently spec'd here since it matches reality. We can always expand if needed, i.e., if non-GUI environments start to do something special here. I'll resolve this but feel free to re-open if you feel strongly.